feat: refactor requireConfig option (#15594)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

## requireConfig

> You can choose from the following behaviors for the `requireConfig` config option:
> 
> - `"required"`: a repository config file required, if config file is missing Renovate will skip the repository gracefully.
> - `"optional"`: if it is optional for the repository to contain a config. if config files exists Renovate will run and use it.
> - `"ignored"`: Renovate will run and ignore any repository config files.

Reviewer: ```suggestion
By default, Renovate needs a Renovate config file in each repository where it runs.
Also by default, Renovate will run on all repositories, even if any repository lacks a Renovate config file.

You can use the `requireConfig` option to change this behavior:

- `"required"`: a repository config file must be present, if there's no config file then Renovate will skip the repository gracefully.
- `"optional"`: it's optional for a repository to have a Renovate config file, if the config file exists, Renovate will use it when it runs
- `"ignored"`: Renovate will run and ignore any repository config files
```

How about something like this?

Author: looks good 
thanks

## requireConfig

By default, Renovate needs a Renovate config file in each repository where it runs.
Also by default, Renovate will run on all repositories, even if any repository lacks a Renovate config file.

You can use the `requireConfig` option to change this behavior:

> - `"required"`: a repository config file must be present, if there's no config file then Renovate will skip the repository gracefully.

Reviewer: ```suggestion
- `"required"`: a repository config file must be present, if there's no config file then Renovate will skip the repository gracefully
```

> By default, Renovate needs a Renovate config file in each repository where it runs.
> Also by default, Renovate will run on all repositories, even if any repository lacks a Renovate config file.
> 
> You can use the `requireConfig` option to change this behavior:
> 
> - `"required"`: a repository config file must be present, if there's no config file then Renovate will skip the repository gracefully
> - `"optional"`: it's optional for a repository to have a Renovate config file, if the config file exists, Renovate will use it when it runs
> - `"ignored"`: Renovate will run and ignore any repository config files

Reviewer 2: ```suggestion
By default, Renovate needs a Renovate config file in each repository where it runs before it will propose any dependency updates.

The configuration possibilities for this option are:

- `"required"` (default): a repository config file must be present.
- `"optional"`: if a config file exists, Renovate will use it when it runs.
- `"ignored"`: config files in the repo will be ignored, and have no effect

This feature is closely related to the `onboarding` config option. The possible combinations of `requireConfig` and `onboarding` are described below:

|                        | onboarding=true                                                                                                                                         | onboarding=false                                                   |
|------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------|
| requireConfig=required | An onboarding PR will be created if no config file exists. If the onboarding PR is closed and there's no config file, then the repository is skipped.   | Repository is skipped unless a config file is added manually.      |
| requireConfig=optional | An onboarding PR will be created if no config file exists. If the onboarding PR is closed and there's no config file, the repository will be processed. | Repository is processed regardless of config file presence or not. |
| requireConfig=ignored  | No onboarding PR will be created and repo will be processed while ignoring any config file present.                                                     | Repository is processed, any config file is ignored.               |
```

## requireConfig

By default, Renovate needs a Renovate config file in each repository where it runs before it will propose any dependency updates.

> The configuration possibilities for this option are:

Reviewer: ```suggestion
You can choose any of these settings:
```

How about something like this?

> - `"required"` (default): a repository config file must be present.
> - `"optional"`: if a config file exists, Renovate will use it when it runs.

Reviewer: ```suggestion
- `"required"` (default): a repository config file must be present
- `"optional"`: if a config file exists, Renovate will use it when it runs
```

We don't use `.` at the end of list items.

- `"ignored"`: config files in the repo will be ignored, and have no effect

This feature is closely related to the `onboarding` config option.
The combinations of `requireConfig` and `onboarding` are:

> |                        | onboarding=true                                                                                                                                         | onboarding=false                                              |
> | ---------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------- |
> | requireConfig=required | An onboarding PR will be created if no config file exists. If the onboarding PR is closed and there's no config file, then the repository is skipped.   | Repository is skipped unless a config file is added manually. |
> | requireConfig=optional | An onboarding PR will be created if no config file exists. If the onboarding PR is closed and there's no config file, the repository will be processed. | Repository is processed regardless of config file presence.   |
> | requireConfig=ignored  | No onboarding PR will be created and repo will be processed while ignoring any config file present.                                                     | Repository is processed, any config file is ignored.          |

Reviewer: ```suggestion
|                        | `onboarding=true`                                                                                                                                         | `onboarding=false`                                              |
| ---------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------- |
| `requireConfig=required` | An onboarding PR will be created if no config file exists. If the onboarding PR is closed and there's no config file, then the repository is skipped.   | Repository is skipped unless a config file is added manually. |
| `requireConfig=optional` | An onboarding PR will be created if no config file exists. If the onboarding PR is closed and there's no config file, the repository will be processed. | Repository is processed regardless of config file presence.   |
| `requireConfig=ignored`  | No onboarding PR will be created and repo will be processed while ignoring any config file present.                                                     | Repository is processed, any config file is ignored.          |
```

Can we try how it looks when we use `monospace` for the settings?

Author: Hi @HonkingGoose 
this is how it looks after your last changes
![image](https://user-images.githubusercontent.com/98313426/169002362-60e208af-9617-4601-84aa-ce0d50674dce.png)
